src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -154,5 +172,37 @@ private void UpdateNextRefreshableSecretFromCache()
                 _nextRefreshTime = null;
             }
         }
+
+        private void UpdateNextRefreshTimeForFailedSecret(string key, CachedKeyVaultSecret cachedSecret)
```

Comment:
This is almost duplicated as `SetSecretInCache`. Any chance we can merge the code? For example, `refreshInterval` will be the same as `refreshInterval.CalculateBackoffTime(cachedSecret.RefreshAttempts)` when `RefreshAttempts` is 1.

Reply:
I did some refactoring to reduce code duplication. However, I don't think we should optimize the calculation of refresh time based on which attempt it is. The code may become unnecessarily complicated.

Reply:
No, not meant to depend on the implementation detail of the calculation, but we can have something like 
```cs
refreshSecretAt = DateTimeOffset.UtcNow.Add(
    cachedSecret.RefreshAttempts == 0 ? refreshInterval : refreshInterval.CalculateBackoffTime(cachedSecret.RefreshAttempts)
);
```
This can further simplify your current `GetCacheExpirationTimeForSecret`.

Reply:
Did more refactoring to get closer to the suggested approach.